doc/good_stv_api_0.1-alpha.md

```diff
@@ -0,0 +1,127 @@
+FORMAT: 1A
+
+# Good STV
+
+Good STV is a simple API allowing a group of people to vote on a minor election,
+such as a student society committee position.
+
+# Group Polls
+
+Resources related to polls in the API.
+
+## Poll collection [/polls]
+
+### Create a poll [POST]
+
+This action creates a new poll. It takes a JSON object containing a title and a list of candidates.
+
+* `poll` (`string`) - The poll name.
+* `choices` (`array[string]`) - The names of each of the candidates.
+
+* Request (application/json)
+
+    ```
+    {
+        "poll": "Redbrick treasurer election.",
+        "choices": [
+            "Alice",
+            "Bob",
+            "Charlie"
+        ]
+    }
+    ```
+
+*   Response 201 (application/json)
+
+    *   Headers
+
+    ```
+    Location: /polls/a1b2c3
+    ```
+
+    *   Body
+
+    ```json
+    {
+        "admin_key": "abcdefg123456"
+    }
+    ```
+
+## Poll [/polls/{id}]
+
+*   Parameters
+    *   `id`: `a1b2c3` (alphanumeric string, required) - Unique identifier for the
```

Comment:
Setting a small limit, 6 characters, because they'll be the codes displayed to the voters. I say we don't worry about collision for right now, but a future solution could be to have 2 separate ids, a long UUID for internal identification, and a shorter one to display to the users. We could enforce that multiple polls can share the same short ID, iff only one of them is marked as in-progress.